src/shared/user-record.c

```diff
@@ -37,19 +37,22 @@ static int parse_alloc_uid(const char *path, const char *name, const char *t, ui
         *ret_uid = uid;
         return 0;
 }
+#endif
 
-static int read_login_defs(login_defs *ret_defs, const char *path) {
-        _cleanup_fclose_ FILE *f = NULL;
+int read_login_defs(login_defs *ret_defs, const char *path, const char *root) {
```

Comment:
the "path" argument is always the same no? maybe drop it, and just take the root?

Reply:
It's used in tests...